### PR TITLE
docs(whitepaper): add related work

### LINKED ISSUE #235: docs(whitepaper): improve related work
**Description**
As a contributor/maintainer/developer I want to have specific information about the related work on the blockchain interoperability space. At this moment, the whitepaper provides little insights on what is happening on other projects.

Acceptance Criteria
The whitepaper exposes different approaches to the blockchain interoperability problem, and how Cactus relates to them

### LINKED ISSUE #236: docs(whitepaper): improve related work
**Description**
As a contributor/maintainer/developer I want to have specific information about the related work on the blockchain interoperability space. At this moment, the whitepaper provides little insights on what is happening on other projects.

**Acceptance Criteria**
The whitepaper exposes different approaches to the blockchain interoperability problem, and how Cactus relates to them

cc @petermetz @jonathan-m-hamilton @takeutak @sfuji822

### PR DESCRIPTION
**Pull request by RafaelAPB**
Adds an introduction about related work on blockchain interoperability, based on a recent academic survey.

Fixes #236

Signed-off-by: Rafael Belchior <rafael.belchior@tecnico.ulisboa.pt>

### CODE REVIEW

File: whitepaper/whitepaper.md [hunk 1]
```diff
@@ -1187,10 +1188,31 @@ Web 3.0 applications (decentralized apps or *DApps*) which interact with blockch
...
+
+![rw](./related-work-categories.png)
+
+Cryptocurrency-directed interoperability approaches identify and define different strategies for chain interoperability across public blockchains, most of them implementing cryptocurrencies.
+
+Blockchain engines are frameworks that provide reusable data, network, consensus, incentive, and contract layers for the creation of customized blockchains, serving general use-cases. Emerging blockchains are, for example, the Cosmos Network and Polkadot. 
+
+The Blockchain Connector category is composed of interoperability solutions that are not cryptocurrency-directed or blockchain engines. Several sub-categories exist: Trusted Relays, Blockchain Agnostic Protocols, Blockchain of Blockchains, and Blockchain Migrators”.
+
+White Hyperledger Cactus has caracteristics from the the three categories, it can be considered a Blockchain Connector (namely a Trusted Relay). In particular, Cactus focuses on providing multiple use case scenarios via a trusted consortium.
```
Comment (petermetz) [inline whitepaper/whitepaper.md:1204]: Typo? `White Hyperledger Cactus`

File: whitepaper/whitepaper.md [hunk 2]
```diff
@@ -1187,10 +1188,31 @@ Web 3.0 applications (decentralized apps or *DApps*) which interact with blockch
...
+
+The Blockchain Connector category is composed of interoperability solutions that are not cryptocurrency-directed or blockchain engines. Several sub-categories exist: Trusted Relays, Blockchain Agnostic Protocols, Blockchain of Blockchains, and Blockchain Migrators”.
+
+White Hyperledger Cactus has caracteristics from the the three categories, it can be considered a Blockchain Connector (namely a Trusted Relay). In particular, Cactus focuses on providing multiple use case scenarios via a trusted consortium. 
+Trusted relays allow the discovery of the target blockchains, appearing often in a permissioned blockchain environment, where cross-blockchain transactions are routed by trusted escrow parties.
+ 
+ However, depending on the validator plugin, the trust on the relay can be decentralized, making Cactus a decentralized, general-purpose, trustless relay. 
+It is possible that Cactus becomes a Blockchain of blockchains, given that it will provide mechanisms for developers to build cross-chain dApps. Conversely, the blockchain migrator feature paves the way for building a solution that performs data migration across blockchains.
```
Comment (petermetz) [inline whitepaper/whitepaper.md:1208]: @RafaelAPB 
`It is possible that Cactus becomes a Blockchain of blockchains`

Never say never for sure, but I think we've stated the opposite a few times so this may confuse the reader. 
Also depends on how the phrase `blockchain of blockchains` is understood, from an abstract, theoretical perspective you could call it a blockchain of blockchains if you just meant by it that one has access to multiple blockchains through a Cactus deployment, but the way we used the term so far is when `blockchain of blockchains` means that there's yet another blockchain storing actual data (blocks) and it's somehow "meta" relative to the rest of the chains.
Long story short, at the moment I'd say we'd rather not provide a full blown implementation of a distributed ledger (blockchain) but just aim to connect existing software that does that.

Also possible that I'll end up looking like a fool in the future for saying all this, but I'm at peace with that. :-)

Comment (RafaelAPB) [reply whitepaper/whitepaper.md:1208]: Yes, agreed! I meant this because, in fact, a blockchain of blockchains might just correspond to a fancy validator plugin, tracking the decentralized voting process.
